mtools/mtransfer/mtransfer.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import re
+import sys
+import bson
+import wiredtiger
+
+from mtools.util.cmdlinetool import BaseCmdLineTool
+from mtools.version import __version__
+
+codec_options = bson.codec_options.CodecOptions(uuid_representation=bson.binary.STANDARD)
+
+
+class MTransferTool(BaseCmdLineTool):
+    def __init__(self):
+        BaseCmdLineTool.__init__(self)
+
+        self.argparser.description = ('Imports and exports databases to/from MongoDB.')
+
+        self.argparser.add_argument('--dbpath', dest='dbpath', default='.', nargs=1,
+                                    help='MongoDB database path')
+
+        self.argparser.add_argument('--force', action='store_true',
+                                    help='ignore safety checks')
+
+        self.argparser.add_argument('--verbose', action='store_true',
+                                    help='enable verbose output')
+
+        self.argparser.add_argument('command', choices=['export', 'import'])
+
+        self.argparser.add_argument('database', nargs=1, type=str,
+                                    help='name of the database to export / import')
+
+    def run(self, arguments=None):
+        BaseCmdLineTool.run(self, arguments)
+
+        self.dbpath = self.args['dbpath']
+        self.force = self.args['force']
+        self.verbose = self.args['verbose']
+
+        # Read storage.bson, sanity check.
+        try:
+            storage_raw = open(os.path.join(self.dbpath, 'storage.bson'), 'r').read()
```

Comment:
`bson.decode` requires a bytes like object. This should be changed so the file is read as bytes:

`storage_raw = open(os.path.join(self.dbpath, 'storage.bson'), 'rb').read()`

Reply:
Actually, this is part of a bigger problem. The program runs fine with python2, but it fails with python3 due to the distinction between bytes and strings.